### PR TITLE
chore: convert icon button stories to CSF3

### PR DESCRIPTION
**Pull request by LucidityDesign**

### CODE REVIEW

File: src/components/controls/buttons/IconButton/IconButton.stories.tsx [hunk 1]
```diff
@@ -1,82 +1,89 @@
...
 import { within } from "@storybook/testing-library"
-import { ComponentProps } from "react"
 import { SnapshotWrapper } from "../../../../lib/storybook/SnapshotWrapper"
 import { IconButton } from "./IconButton"
 
 export default {
   title: "Controls/Buttons/IconButton",
   component: IconButton,
   argTypes: {
+    // TODO set type
```
Comment (LucidityDesign) [inline src/components/controls/buttons/IconButton/IconButton.stories.tsx:11]: @filiptammergard do you know how to handle the missing type?

Comment (filiptammergard) [reply src/components/controls/buttons/IconButton/IconButton.stories.tsx:11]: Oh! `Meta` is recommended from v7.0, but let's use `ComponentMeta` for now instead!

File: src/components/controls/buttons/IconButton/IconButton.stories.tsx [hunk 2]
```diff
@@ -13,70 +12,77 @@ export default {
...
 export const Snapshot = (): JSX.Element => (
   <SnapshotWrapper>
-    {[Basic, EllipsisButton, Primary, Tertiary, Disabled].map((Story, index) => (
-      // eslint-disable-next-line react/no-array-index-key
-      <Story key={index} {...(Story.args as ComponentProps<typeof IconButton>)} />
-    ))}
+    {[Basic, EllipsisButton, Primary, Tertiary, Disabled].map((Story, index) => {
+      return (
+        // eslint-disable-next-line react/no-array-index-key
+        <IconButton key={index} {...Story.args} />
```
Comment (filiptammergard) [inline src/components/controls/buttons/IconButton/IconButton.stories.tsx:83]: Shouldn't this be `<Story />` instead of `<IconButton>`?

**Review by filiptammergard — APPROVED** [review]

File: src/components/controls/buttons/IconButton/IconButton.stories.tsx (hunk 2 above)
Comment (LucidityDesign) [reply src/components/controls/buttons/IconButton/IconButton.stories.tsx:83]: `Story` is the CSF3 object and not a functional component anymore

**Pull request merged**